### PR TITLE
Add Refchecker to Tests

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

* Fix `utils/refchecker.py`
* Add `utils/refchecker.py` checks for RHEL7 and RHEL8 CIS profiles

#### Rationale:

* Make `utils/refchecker.py` useful
* Help find bugs sooner

#### Review Hints:
* Remove the CIS keys from a file like `linux_os/guide/system/software/integrity/crypto/configure_crypto_policy/rule.yml`, and it should make the tests fail.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -196,3 +196,65 @@ add_test(
...
 endif()
+
+if (PYTHON3_EXECUTABLE AND SSG_PRODUCT_RHEL8)
+    add_test(
+        NAME "refchecker-rhel8-cis_workstation_l1"
+        COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/utils/refchecker.py" --json "${CMAKE_SOURCE_DIR}/build/rule_dirs.json" --build-config-yaml "${CMAKE_BINARY_DIR}/build_config.yml" "rhel8" "cis_workstation_l1" "cis"
+    )
+    set_tests_properties("refchecker-rhel8-cis_workstation_l1" PROPERTIES FIXTURES_REQUIRED "rule-dir-json")
+    set_tests_properties("refchecker-rhel8-cis_workstation_l1" PROPERTIES DEPENDS "test-rule-dir-json")
+    set_tests_properties("refchecker-rhel8-cis_workstation_l1" PROPERTIES LABELS quick)
```
Comment (jan-cerny) [inline tests/CMakeLists.txt:207]: Can we create a macro for the tests that call refchecker.py? That would reduce the repeating code.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -196,3 +196,65 @@ add_test(
...
+    add_test(
+        NAME "refchecker-rhel8-cis_server_l1"
+        COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/utils/refchecker.py" --json "${CMAKE_SOURCE_DIR}/build/rule_dirs.json" --build-config-yaml "${CMAKE_BINARY_DIR}/build_config.yml" "rhel8" "cis_server_l1" "cis"
+    )
+    set_tests_properties("refchecker-rhel8-cis_server_l1" PROPERTIES FIXTURES_REQUIRED "rule-dir-json")
+    set_tests_properties("refchecker-rhel8-cis_server_l1" PROPERTIES DEPENDS "test-rule-dir-json")
+    set_tests_properties("refchecker-rhel8-cis_server_l1" PROPERTIES LABELS quick)
+    add_test(
+        NAME "refchecker-rhel8-cis_server_l2"
+        COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/utils/refchecker.py" --json "${CMAKE_SOURCE_DIR}/build/rule_dirs.json" --build-config-yaml "${CMAKE_BINARY_DIR}/build_config.yml" "rhel8" "cis_server_l1" "cis"
```
Comment (jan-cerny) [inline tests/CMakeLists.txt:224]: You test `cis_server_l1` for RHEL 8 for the second time, you want to test server profile level 2 instead which has just `cis` profile ID.

File: utils/refchecker.py [hunk 3]
```diff
@@ -16,19 +17,22 @@
 import ssg.utils
 
 SSG_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+RULES_JSON = os.path.join(SSG_ROOT, "build", "rule_dirs.json")
+BUILD_CONFIG = os.path.join(SSG_ROOT, "build", "build_config.yml")
+CONTROLS_DIR = os.path.join(SSG_ROOT, 'controls')
```
Comment (jan-cerny) [inline utils/refchecker.py:22]: We probably should use double quotes because we use them in all other string literals in this script.

**Review by jan-cerny — CHANGES_REQUESTED** [review]
I have run some of the tests on the RHEL 8 content and when I intentionally removed some CIS references from rules the test started to fail as is expected.

**Comment by marcusburghardt** [comment]
@jan-cerny , I saw you already reviewed this PR once. Are you comfortable to assign it to yourself?

**Review by jan-cerny — APPROVED** [review]
I have run the tests locally and if I break something they start to fail as expected. Also, I have seen that they run in GitHub Actions.

**Pull request merged**